p4-16/psa/PSA.mdk

```diff
@@ -338,46 +346,43 @@ packet contents and metadata when a packet begins ingress processing.
 |--------------|-------------|---------------|--------------|-----|
 ~
 
-TBD: Add PORT_RECIRCULATE near PORT_CPU in psa.p4
-
-Note that the ingress_port value for a resubmitted packet could be
-PORT_RECIRCULATE if a packet was recirculated, and then that
+Note that the `ingress_port` value for a resubmitted packet could be
+`PORT_RECIRCULATE` if a packet was recirculated, and then that
 recirculated packet was resubmitted.
 
-### packet_in for NFP and NFCPU packets
+### Initial packet contents for packets from ports
 
-For Ethernet ports, packet_in contains the Ethernet frame starting
-with the Ethernet header.  It does not include the Ethernet frame CRC.
-TBD whether the payload is always the minimum of 46 bytes (64 byte
-minimum Ethernet frame size, minus 14 bytes of header, minus 4 bytes
-of CRC), or whether an implementation is allowed to leave some of
-those bytes out.
+For Ethernet ports, `packet_in` for FP and NFCPU path packets contains
+the Ethernet frame starting with the Ethernet header.  It does not
+include the Ethernet frame CRC.  TBD whether the payload is always the
```

Comment:
Given that we're planning for a programmable switch and even though most applications are likely to include Ethernet, I don't see a strong need to constrain the minimum packet size for PSA. It is likely to be target dependent in any case. And if one wants to build a custom protocol in which the entire packet consists of headers, we should not prevent it. There are target limits on how big a header can be as well.